Mathlib/AlgebraicGeometry/EllipticCurve/Affine.lean

```diff
@@ -188,6 +188,9 @@ lemma eval_polynomial_zero : (W.polynomial.eval 0).eval 0 = -W.a₆ := by
   simp only [← C_0, eval_polynomial, zero_add, zero_sub, mul_zero, zero_pow <| Nat.succ_ne_zero _]
 #align weierstrass_curve.eval_polynomial_zero WeierstrassCurve.Affine.eval_polynomial_zero
 
+lemma map_polynomial {S} [CommRing S] (f : R →+* S) :
+    polynomial (W.map f) = W.polynomial.map (mapRingHom f) := by simp [polynomial]
+
```

Comment:
I think since `WeierstrassCurve.Affine` is just an abbrev of `WeierstrassCurve`, it's not defeq abuse even if we don't insert `toAffine`. I think you introduce the abbrev just to be able to use dot notation? Because there's no syntax in Lean 4 that allows using dot notation inside a sub-namespace.

Of course if e.g. both `Affine` and `Jacobian` namespaces are open, then Lean could get confused which `polynomial` you are referring to, but in that case I still think both `(W.map φ).toAffine.polynomial` and `Affine.polynomial (W.map φ)` are acceptable solutions.

Reply:
I introduced the `abbrev` mostly to avoid the clashes for `Affine`/`Projective`/`Jacobian`, and I think both are fine. My comment here is just for the relocation of the lemma.